Automattic/jetpack
# External Media: Ensure the correct connect URL is used.

A fix to the blog nonce check on WPCOM highlighted that the connect URL retrieved while connecting to services like Google Photos, had the incorrect blog ID (when a user has multiple blogs) and the nonce was based on the public-api blog ID.

This fixes that problem by swapping to using the site specific endpoint to get the service details.

### Other information:


## Jetpack product discussion
p1717415328965419-slack-C02JJ910CNL

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* With a connected Jetpack site, insert a media block like an image
* Click the Select Image option and from the dropdown Google Photos
* You'll see a screen to connect to Google Photos, click the button
* A window will pop up to begin the auth process
* Without this change you will receive an Invalid nonce error
* With this change, you should be able to create a connection to Google Photos
